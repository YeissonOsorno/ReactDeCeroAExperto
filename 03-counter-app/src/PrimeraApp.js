import React from 'react';
import PropTypes from 'prop-types';

const PrimerApp = ({saludo,subtitulo})=>{
    //Antes del return puedo poner cualquier codigo JavaScript
    
    return (
        <>
            <h1>{saludo}!!</h1>
            {/*<pre>{JSON.stringify(greet,null,3)}</pre>*/}
            <p>{subtitulo}</p>
        </>
    )

}
//Obligamos a los devs que envien los properties a nuectro componente
PrimerApp.propTypes ={
    saludo:PropTypes.string.isRequired
}

//Default Props
PrimerApp.defaultProps={
    subtitulo:'Soy un Subtitulo'
}
export default PrimerApp;