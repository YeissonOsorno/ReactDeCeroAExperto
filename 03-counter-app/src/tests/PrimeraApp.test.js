//import {render} from '@testing-library/react';
import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <primeraApp/>',()=>{
    /*test('Debe de mostrar el mesanje "Hola soy Goku"', () => {
        const saludo ="Hola Soy Goku";
        const wrapper = render(<PrimeraApp saludo={saludo}/>)

        expect(wrapper.getByText(saludo)).toBeInTheDocument();
    })*/

    test('Debe de mostrar el componente bien cargado"', () => {
        const saludo = 'Hola Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola Soy Goku';
        const subtitulo = 'El Rey de los Sayayines';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>)

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
        
    })
    
    
})