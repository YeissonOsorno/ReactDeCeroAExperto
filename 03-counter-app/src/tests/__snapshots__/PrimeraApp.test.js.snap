// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <primeraApp/> Debe de mostrar el componente bien cargado" 1`] = `
<Fragment>
  <h1>
    Hola Soy Goku
    !!
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
